readme.md

```diff
@@ -45,11 +45,55 @@ Number of milliseconds to wait after `condition` resolves to `false` before call
 
 ##### timeout
 
-Type: `number`\
+Type: `number | TimeoutOptions`\
 Default: `Infinity`
 
 Number of milliseconds to wait before automatically rejecting with a `TimeoutError`.
 
+You can customize the `TimeoutError` by specifying `TimeoutOptions` instead of `number`.
+
+```js
+import pWaitFor from 'p-wait-for';
+import {pathExists} from 'path-exists';
+
+await pWaitFor(() => pathExists('unicorn.png'), {
+	timeout: {
+		milliseconds: 100,
+		message: MyError('Time’s up!'),
+		customTimers: {
+			setTimeout: requestAnimationFrame
+		}
+	}
+});
+console.log('Yay! The file now exists.');
+```
```

Comment:
Readme and index.d.ts should be in sync. This example is missing from index.d.ts.